task-loops/pkg/reconciler/tasklooprun/tasklooprun.go

```diff
@@ -178,93 +193,74 @@ func (c *Reconciler) reconcile(ctx context.Context, run *v1alpha1.Run, status *t
 		return nil
 	}
 
-	// Update status of TaskRuns.  Return the TaskRun representing the highest loop iteration.
-	highestIteration, highestIterationTr, err := c.updateTaskRunStatus(logger, run, status)
+	// Update the status of the TaskRuns created from this Run on prior reconciliations.
+	// updateTaskRunStatus() also handles TaskRun cancellation and retry.
+	totalRunning, highestIteration, taskRunFailed, err := c.updateTaskRunStatus(ctx, logger, run, status, taskLoopSpec)
```

Comment:
Minor comment:
Naming is hard  and I prefer this to the previous naming. It might still be helpful to add a small comment explaining what is meant here - e.g. `highestIteration` refers to number of taskruns which have been started until now.

Reply:
done